iphone/cli/hooks/frameworks.js

```diff
@@ -0,0 +1,841 @@
+		this._projectUuid = xcodeProject.hash.project.rootObject;
+		this._pbxProject = this._xobjs.PBXProject[this._projectUuid];
+		this._mainTargetUuid = this._pbxProject.targets.filter((target) => {
+			return target.comment.replace(/^"/, '').replace(/"$/, '') === this._builder.tiapp.name;
+		})[0].value;
+		this._mainTarget = this._xobjs.PBXNativeTarget[this._mainTargetUuid];
+		this._mainGroupChildren = this._xobjs.PBXGroup[this._pbxProject.mainGroup].children;
+		this._frameworksGroup = this._xobjs.PBXGroup[this._mainGroupChildren.filter((child) => {
+			return child.comment === 'Frameworks';
+		})[0].value];
+		this._frameworksBuildPhase = this._xobjs.PBXFrameworksBuildPhase[this._mainTarget.buildPhases.filter((phase) => {
+			return this._xobjs.PBXFrameworksBuildPhase[phase.value];
+		})[0].value];
+
+		this._frameworkSearchPaths = new Map();
+		this._runpathSearchPaths = new Map();
+		let buildConfigurations = this._xobjs.XCConfigurationList[this._mainTarget.buildConfigurationList].buildConfigurations;
+		for (let buildConf of buildConfigurations) {
+			if (!this._frameworkSearchPaths.has(buildConf.value)) {
+				this._frameworkSearchPaths.set(buildConf.value, new Set());
+				this._runpathSearchPaths.set(buildConf.value, new Set());
+			}
+
+			let buildSettings = this._xobjs.XCBuildConfiguration[buildConf.value].buildSettings;
+			if (Array.isArray(buildSettings.FRAMEWORK_SEARCH_PATHS)) {
+				let frameworkSearchPaths = this._frameworkSearchPaths.get(buildConf.value);
+				for (let frameworkSearchPath of buildSettings.FRAMEWORK_SEARCH_PATHS) {
+					let cleanFrameworkSearchPath = frameworkSearchPath.replace('"', '');
+					if (!frameworkSearchPaths.has(cleanFrameworkSearchPath)) {
+						frameworkSearchPaths.add(cleanFrameworkSearchPath);
+					}
+				}
+			}
+
+			if (Array.isArray(buildSettings.LD_RUNPATH_SEARCH_PATHS)) {
+				let runpathSearchPaths = this._runpathSearchPaths.get(buildConf.value);
+				for (let runpathSearchPath of buildSettings.LD_RUNPATH_SEARCH_PATHS) {
+					let cleanRunpathSearchPath = runpathSearchPath.replace(/"/g, '');
+					if (!runpathSearchPaths.has(cleanRunpathSearchPath)) {
+						runpathSearchPaths.add(cleanRunpathSearchPath);
+					}
+				}
+			}
+		}
+	}
+
+	/**
+	 * Integrates a frameworks into the Xcode project by adding the required
+	 * build phases and adjusting the framework search path
+	 *
+	 * @param {FrameworkInfo} frameworkInfo Framework metadata
+	 */
+	integrateFramework(frameworkInfo) {
+		let fileRefUuid = this.addFrameworkFileReference(frameworkInfo);
+		this.addLinkFrameworkBuildPhase(frameworkInfo, fileRefUuid);
+		if (frameworkInfo.type === FRAMEWORK_TYPE_DYNAMIC) {
+			this.addEmbeddFrameworkBuildPhase(frameworkInfo, fileRefUuid);
+		}
+		this.addFrameworkSearchPath(path.dirname(frameworkInfo.path));
+	}
+
+	/**
+	 * Add the framework as a new file reference to the Xcode project
+	 *
+	 * @param {frameworkInfo} frameworkInfo Framework metadata
```

Comment:
I believe the type should be `FrameworkInfo`. There are a couple other places below with the same issue.